googlesource/googlesource.go

```diff
@@ -192,6 +190,13 @@ func (s *Source) ReadGoogleRow(ctx context.Context) (err error) {
 	return
 }
 
+func matchColumnName(dataName, columnName string, data sdk.StructuredData) (key string) {
```

Comment:
What is the method supposed to do?

Reply:
matchColumnName matches if the column name is equal to the user defined primary key column. if it is so assign this column name data to key for the record.
Created another function because of complexity.